File: open_grasshopper.py
from rhinomcp import get_rhino_connection

def open_grasshopper():
    """
    Open Grasshopper in Rhino using the MCP connection.
    """
    rhino = get_rhino_connection()
    
    print("Opening Grasshopper...")
    result = rhino.send_command("open_grasshopper", {})
    print("Result:", result)

if __name__ == "__main__":
    open_grasshopper() 

File: rhino_mcp_plugin/Functions/Grasshopper/OpenGrasshopper.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using Newtonsoft.Json.Linq;
using Rhino;
using Rhino.DocObjects;
using Rhino.Geometry;
using rhinomcp.Serializers;
using Rhino.PlugIns;
using Rhino.Commands;

namespace RhinoMCPPlugin.Functions;

public partial class RhinoMCPFunctions
{
    public JObject OpenGrasshopper(JObject parameters)
    {
        try
        {
            // Execute the Grasshopper command
            bool success = RhinoApp.RunScript("Grasshopper", false);
            
            if (success)
            {
                return new JObject
                {
                    ["status"] = "success",
                    ["message"] = "Grasshopper opened successfully"
                };
            }
            else
            {
                return new JObject
                {
                    ["status"] = "error",
                    ["message"] = "Failed to open Grasshopper"
                };
            }
        }
        catch (Exception e)
        {
            return new JObject
            {
                ["status"] = "error",
                ["message"] = e.Message
            };
        }
    }
} 